changelog/fragments/ansible-scaffolding-changes.yaml

```diff
@@ -0,0 +1,24 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      Ansible scaffolding got modified in order to incorporate with Kubebuilder `--component-config` flag changes. Previously, it was scaffolding and doing some file changes without considering the `--component-config` flag.
+      Now, this PR made changes that will consider the `--component-config` flag and modify the scaffolded file.
+
+      1. If the `--component-config` flag is passed then look for `- /manager` in the file in order to add leader election id to the manager.yaml file. If flag is not passed then look for `--leader-elect`.
+      2. Same logic for manager proxy patch file, look for `memory: 64Mi` if the flag is passed. If a flag is not passed then look for `--leader-elect`
+      3. Repeat the same logic for replacing the port number from the file.
+      4. At the end, update the `manager.yaml` file to avoid lint errors.
+    kind: "change"
+    breaking: false
+    migration:
+      header: this change also modified ansible molecule test that was looking for `--leader-elect` in the file.
+      body: |
+        In the advance molecule test, modified code to behave properly with new `--component-config` flag changes.
+
+        ```
+          	err = kbutil.InsertCode(
+                    filepath.Join(ma.ctx.Dir, "config", "default", "manager_auth_proxy_patch.yaml"),
+                    "- \"--leader-election-id=advanced-molecule-operator\"",
+                    managerAuthArgs)
```

Comment:
Isn't the advanced molecule test part of our internal hack script? This does not have any bearing on the users ansible project.

The purpose of the migration is to signify to the user what they would have to change in their project to match a newly scaffolded project.

For example, let's say in previous version we output two files. A foobar.yaml and a baz.yaml.

Now let's say in this PR, we removed `baz.yaml` and we changed the contents of the `foobar.yaml` by adding the following line `- /manager`. 

So what you would say in the migration step is:
1. Remove the baz.yaml it is no longer needed with this version of the SDK
2. After section SOMETHING, add `- /manager`.

The reason we tell them to do this is so that in the future if there were changes to `create api` or other plugins that look at the contents of those files they can see files that look as similar to what the new scaffolds look like.

Reply:
removed this code snippet as it is part of testdata generation.

```
    migration:
      header: this change also modified ansible molecule test that was looking for `--leader-elect` in the file.
      body: |
        In the advance molecule test, modify the below code to behave properly with new `--component-config` flag changes.
        
        ```go
          	"- \"--leader-election-id=advanced-molecule-operator\"",
        ``` 
```

Reply:
got it. thank you for a detailed explanation. :)